src/shared/internal-calendar/internal-calendar.tsx

```diff
@@ -59,6 +60,7 @@ export const Component = (
     const [currentView, setCurrentView] = useState<View>("default");
     const [selectedStartDate, setSelectedStartDate] = useState<string>(); // YYYY-MM-DD
     const [selectedEndDate, setSelectedEndDate] = useState<string>(); // YYYY-MM-DD
+    const [viewCalendarDate, setViewCalendarDate] = useState<Dayjs>();
```

Comment:
I feel `calendarDate` is sufficient also. 

An issue happened when it selected the same month after clicked prev/next in month calendar. So, i created new state to solve this issue.

The calendar flow with `viewCalendarDate`:
1. date is selected.
2. open month calendar
3. clicked prev/next arrow (call below function)
4. setCalendarDate called and still in month calendar (solved above mentioned issue). Also, update the year number in month calendar
  a. not selected the month because in `generateMonthStatus` not using `calendarDate.isSame(date, 'month')`
5. click button or the dropdown to go back day calendar
  a. click Cancel button -> reset value focused value in day calendar
  b. click Done button -> show selected month value in day calendar 
  c. click year dropdown in month calendar -> show selected month value in day calendar 


```
  const handleRightArrowClick = () => {
      switch (currentView) {
          case "default":
              setViewCalendarDate((date) => date.add(1, "month"));
              setCalendarDate((date) => {
                  const nextMonth = date.add(1, "month");
                  performOnCalendarDate(nextMonth);

                  return nextMonth;
              });
              break;
          case "month-options":
              setCalendarDate((date) => date.add(1, "year")); // called me in number 3. 
              break;
          case "year-options":
              setCalendarDate((date) => date.add(10, "year")); // called me in number 3
              break;
      }
  };

```

Reply:
ok I see what you mean, `calendarDate` controls the current view, so you can't use it to calculate the `selected-month` variant. can't use `selectedStartDate` either since there's no initial selection. so a separate `viewCalendarDate` is needed.